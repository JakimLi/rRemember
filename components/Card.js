import React, { Component } from 'react';
import images from '../images/images';
import {
  StyleSheet,
  Image
} from 'react-native';

const styles = StyleSheet.create({
  card: {
    height: 350,
    width: 250,
    margin: 30
  }
});

export default ({card, status}) => {
  return (
    <Image
      style={styles.card} key={card.color + card.point}
      source={status === 'checking' ? images['back'] : images[card.color + card.point]} />
  );
}
