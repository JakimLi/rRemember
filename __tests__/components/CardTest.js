import 'react-native';
import React from 'react';
import Card from '../../components/Card';

import renderer from 'react-test-renderer';

it('it renders correctly', () => {
  const card = {
    color: 'Spades',
    point: 'A'
  }

  const cardTree = renderer.create(
    <Card card={card}/>
  );

  expect(cardTree).toMatchSnapshot();
});
