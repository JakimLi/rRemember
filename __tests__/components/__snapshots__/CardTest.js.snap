exports[`test it renders correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 350,
      "margin": 30,
      "width": 250,
    }
  } />
`;
